src/changeKeys.js

```diff
@@ -1,124 +1,23 @@
-        return result + button.replace("(", "*(");
-      }
-    }
-  }
   if (result === "") {
     if ("+-(0123456789".includes(button)) {
       return button;
     }
-    if (".".includes(button)) {
-      return "0.";
-    }
-  }
-  if ("0123456789".includes(result[result.length - 1])) {
-    const numberOfOpenP = (result.match(/\(/g) || []).length;
-    const numberOfCloseP = (result.match(/\)/g) || []).length;
-    if (
-      "*/-+0123456789.".includes(button) ||
-      (button.key === ")" && numberOfOpenP > numberOfCloseP)
-    ) {
-      return result + button;
-    } else {
-      return result;
-    }
-  }
-  if (result[result.length - 1] === "(") {
-    if ("+-(0123456789".includes(button)) {
-      return result + button;
-    }
-    if (".".includes(button)) {
-      return result + "0.";
-    }
-  }
-  if (result[result.length - 1] === ")") {
-    const numberOfOpenP = (result.match(/\(/g) || []).length;
-    const numberOfCloseP = (result.match(/\)/g) || []).length;
-    if ("/*-+".includes(button)) {
-      return result + button;
-    }
-    if (")".includes(button) && numberOfOpenP > numberOfCloseP) {
-      return result + button;
-    }
   }
-
-  //QS: will we run into a problem in differentiating an operator from a sign? */
-  //For  + / , how do you know it's an operator, or a sign (7--)
-  // You need to watch what is before
-  // digit, ) -> operator
-  // operator, ( -> sign
-  // . -> cannot
-
-  // After an operator, you can only have
-  // - Digit
-  // - sign
-  // - (
-  // - ., but we need to add a zero before
-
-  if ("0123456789)".includes(result[result.length - 2])) {
-    //checking "digit, ) -> operator"
-    if ("/*+-".includes(result[result.length - 1])) {
-      if ("+-(0123456789".includes(button)) {
-        return result + button;
-      }
-      if (".".includes(button)) {
-        return result + "0.";
-      }
-      if ("/".includes(button) && result[result.length - 1] === "-") {
-        return setCharAt(result, result.length - 1, "/");
-      }
-      if ("/".includes(button) && result[result.length - 1] === "+") {
-        return setCharAt(result, result.length - 1, "/");
-      }
-      if ("*".includes(button) && result[result.length - 1] === "-") {
-        return setCharAt(result, result.length - 1, "*");
-      }
-      if ("*".includes(button) && result[result.length - 1] === "+") {
-        return setCharAt(result, result.length - 1, "*");
-      }
-    }
-  }
-  if ("/*+-(".includes(result[result.length - 2])) {
-    if ("-+".includes(result[result.length - 1])) {
-      if ("0123456789(".includes(button)) {
-        return result + button;
-      }
-      if (".".includes(button)) {
-        return result + "0.";
-      }
-    }
-  }
-  if ((result[result.length - 1] || "").includes(".")) {
-    if ("0123456789".includes(button)) {
-      return result + button;
-    }
+  if (lastCharIsNum) {
+    return ([result, button] = lastCharNumber(result, button));
```

Comment:
@grimxreaper what do you think this will return?

Reply:
I was thinking this would return both result, and button separately.

Reply:
How?

Reply:
I'm assuming you are referencing this line by the way since this is what is most in view:
`return ([result, button] = lastCharNumber(result, button));`
I thought with destructuring, that it would return result and button separately.

Reply:
What does it mean to return them separately?

Reply:
Both values are returned instead of only returning one value. So technically the array is returned, and the array elements are assigned to their corresponding variables in the place where they are referenced.